src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -252,6 +254,54 @@ def visit_Assign(self, node: foast.Assign, **kwargs) -> foast.Assign:
         new_target = self.visit(node.target, refine_type=new_value.type, **kwargs)
         return foast.Assign(target=new_target, value=new_value, location=node.location)
 
+    def visit_TupleTargetAssign(
+        self, node: foast.TupleTargetAssign, **kwargs
+    ) -> foast.TupleTargetAssign:
+
+        TargetType = list[foast.Starred | foast.Symbol]
+        values = self.visit(node.value, **kwargs)
+
+        if isinstance(values.type, TupleType):
+            num_elts: int = len(values.type.types)
+            targets: TargetType = node.targets
+            indices: tuple[tuple[int] | int] = compute_assign_indices(targets, num_elts)
+
+            if not any(isinstance(i, tuple) for i in indices) and len(indices) != num_elts:
+                raise FieldOperatorTypeDeductionError.from_foast_node(
+                    node, msg=f"Too many values to unpack (expected {len(indices)})."
+                )
+
+            new_targets: TargetType = []
+            for i, index in enumerate(indices):
+                old_target = targets[i]
+
+                if isinstance(index, tuple):
+                    lower, upper = index
+                    new_type = TupleType(types=[t for t in values.type.types[lower:upper]])
+                    new_target = foast.Starred(
+                        id=foast.DataSymbol(
+                            id=old_target.id.id, location=old_target.location, type=new_type
```

Comment:
```suggestion
                            id=old_target.id, location=old_target.location, type=new_type
```
Nitpicking.

Reply:
This is necessary or Eve will complain of an invalid data type when creating a `DataSymbol`

Reply:
Strange, can you post the error message here as this suggests a bug in eve. `old_target.id` is a `SymbolName` so it should be fine for the requested `Coerced[SymbolName]`.

Reply:
```python
tests/functional_tests/ffront_tests/test_execution.py:1323 (test_tuple_unpacking_star_multi[executor])
value = Symbol(location=SourceLocation(line=1348, column=5, source='/home/sk/Dev/gt4py/tests/functional_tests/ffront_tests/tes...DeferredSymbolType(constraint=<class 'functional.ffront.common_types.DataType'>), namespace=<Namespace.LOCAL: 'local'>)

    def _type_converter(value: Any) -> _T:
        try:
            return (
                value
                if isinstance(value, type_annotation)  # type: ignore[arg-type]
>               else type_annotation(value)  # type: ignore
            )

../../../src/eve/datamodels/core.py:1007: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'eve.concepts.SymbolName'>
value = Symbol(location=SourceLocation(line=1348, column=5, source='/home/sk/Dev/gt4py/tests/functional_tests/ffront_tests/tes...DeferredSymbolType(constraint=<class 'functional.ffront.common_types.DataType'>), namespace=<Namespace.LOCAL: 'local'>)

    def __new__(cls, value: str) -> ConstrainedStr:
        if cls is ConstrainedStr:
            raise TypeError(f"{cls} cannot be directly instantiated, it should be subclassed.")
        if not isinstance(value, str) or not cls.regex.fullmatch(value):
>           raise ValueError(
                f"{cls.__name__}('{value}') does not satisfies RE constraint {cls.regex}."
            )
E           ValueError: SymbolName('a__0') does not satisfies RE constraint re.compile('^[a-zA-Z_]\\w*$').

../../../src/eve/type_definitions.py:120: ValueError

The above exception was the direct cause of the following exception:

fieldview_backend = <function executor at 0x7f20e3336a70>

    def test_tuple_unpacking_star_multi(fieldview_backend):
        size = 10
        inp = np_as_located_field(IDim)(np.ones((size)))
        out = tuple(np_as_located_field(IDim)(np.ones(size) * i) for i in range(3 * 4))
    
        OutType = tuple[
            Field[[IDim], float64],
            Field[[IDim], float64],
            Field[[IDim], float64],
            Field[[IDim], float64],
            Field[[IDim], float64],
            Field[[IDim], float64],
            Field[[IDim], float64],
            Field[[IDim], float64],
            Field[[IDim], float64],
            Field[[IDim], float64],
            Field[[IDim], float64],
            Field[[IDim], float64],
        ]
    
        @field_operator
>       def unpack(
            inp: Field[[IDim], float64],
        ) -> OutType:

test_execution.py:1345: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../src/functional/ffront/decorator.py:640: in field_operator
    return field_operator_inner if definition is None else field_operator_inner(definition)
../../../src/functional/ffront/decorator.py:638: in field_operator_inner
    return FieldOperator.from_function(definition, backend)
../../../src/functional/ffront/decorator.py:466: in from_function
    foast_definition_node = FieldOperatorParser.apply(source_def, closure_vars, annotations)
../../../src/functional/ffront/dialect_parser.py:101: in apply
    output_ast = cls._postprocess_dialect_ast(
../../../src/functional/ffront/func_to_foast.py:106: in _postprocess_dialect_ast
    foast_node = FieldOperatorTypeDeduction.apply(foast_node)
../../../src/functional/ffront/foast_passes/type_deduction.py:164: in apply
    typed_foast_node = cls().visit(node)
../../../src/eve/traits.py:162: in visit
    result = super().visit(node, **kwargs)
../../../src/eve/visitors.py:123: in visit
    return visitor(node, **kwargs)
../../../src/functional/ffront/foast_passes/type_deduction.py:172: in visit_FunctionDefinition
    new_body = self.visit(node.body, **kwargs)
../../../src/eve/traits.py:162: in visit
    result = super().visit(node, **kwargs)
../../../src/eve/visitors.py:123: in visit
    return visitor(node, **kwargs)
../../../src/eve/visitors.py:175: in generic_visit
    return node.__class__(  # type: ignore
../../../src/eve/visitors.py:178: in <genexpr>
    if (new_child := self.visit(child, **kwargs)) is not NOTHING
../../../src/eve/traits.py:162: in visit
    result = super().visit(node, **kwargs)
../../../src/eve/visitors.py:123: in visit
    return visitor(node, **kwargs)
../../../src/functional/ffront/foast_passes/type_deduction.py:282: in visit_TupleTargetAssign
    id=foast.DataSymbol(
/usr/lib/python3.10/typing.py:957: in __call__
    result = self.__origin__(*args, **kwargs)
<attrs generated init functional.ffront.field_operator_ast.Symbol>:3: in __init__
    _setattr(self, 'id', __attr_converter_id(id))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

value = Symbol(location=SourceLocation(line=1348, column=5, source='/home/sk/Dev/gt4py/tests/functional_tests/ffront_tests/tes...DeferredSymbolType(constraint=<class 'functional.ffront.common_types.DataType'>), namespace=<Namespace.LOCAL: 'local'>)

    def _type_converter(value: Any) -> _T:
        try:
            return (
                value
                if isinstance(value, type_annotation)  # type: ignore[arg-type]
                else type_annotation(value)  # type: ignore
            )
        except Exception as error:
>           raise TypeError(
                f"Error during coertion of given value '{value}' for field '{name}'."
            ) from error
E           TypeError: Error during coertion of given value 'a__0' for field 'Symbol.id'.

../../../src/eve/datamodels/core.py:1010: TypeError


```